src/Tempest/Http/HttpApplication.php

```diff
@@ -16,33 +17,29 @@
 
 final readonly class HttpApplication implements Application
 {
-    public function __construct(
-        private Container $container,
-        private AppConfig $appConfig,
-    ) {
-    }
+    public function __construct(private Container $container) {}
 
-    public static function boot(string $root, ?AppConfig $appConfig = null): self
+    /** @param \Tempest\Core\DiscoveryLocation[] $discoveryLocations */
+    public static function boot(
+        string $root,
+        array $discoveryLocations = [],
```

Comment:
In theory, this can move to a discovery configuration file. It's "framework dependency" so it's config will be in the application skeleton or autowired with defaults.

Reply:
I agree, it's also more than $discoveryLocations, it's locations, classes, and whether cache is enabled or not

Reply:
I might disagree with my previous agreement, but I need to sleep on the matter

Reply:
That can all belong in a config file. 😉

Reply:
I've slept. I think there's a good argument to be made against a config file: config files are discovered by Tempest, except this one. That would cause a lot of confusion once again as before. I like the current approach better.

Reply:
So if I as a developer want to adjust default discovery locations, I'm doing this by modifying both the `tempest` and `public/index.php` files?